main/src/cgeo/geocaching/CacheDetailActivity.java

```diff
@@ -2007,6 +2018,32 @@ public View getView(final int position, final View convertView, @NonNull final V
                 final CachedetailWaypointsHeaderBinding headerBinding = CachedetailWaypointsHeaderBinding.inflate(getLayoutInflater(), v, false);
                 v.addHeaderView(headerBinding.getRoot());
 
+                final CachedetailWaypointsCheckboxBinding checkboxBinding = CachedetailWaypointsCheckboxBinding.inflate(getLayoutInflater(), v, false);
+                v.addHeaderView(checkboxBinding.getRoot());
+
+                checkboxBinding.hideVisitedWaypoints.setOnCheckedChangeListener(new CompoundButton.OnCheckedChangeListener() {
+                    @Override
+                    public void onCheckedChanged(final CompoundButton buttonView, final boolean isChecked) {
+                        hideVisitedWaypoints = isChecked;
+
+                        final List<Waypoint> sortedWaypoints2 = new ArrayList<>(cache.getWaypoints());
+                        final Iterator<Waypoint> waypointIterator2 = sortedWaypoints2.iterator();
+                        while (waypointIterator2.hasNext()) {
+                            final Waypoint waypointInIterator = waypointIterator2.next();
+                            if (waypointInIterator.isVisited() && hideVisitedWaypoints) {
+                                waypointIterator2.remove();
+                            }
+                        }
+                        Collections.sort(sortedWaypoints2, cache.getWaypointComparator());
+                        //ActivityMixin.showShortToast(activity, "Aantal : " + sortedWaypoints.size());
```

Comment:
We have this code part twice, so maybe it would make sense to move it into a generalized method returning the filtered and sorted list.

Oh, and please remove commented out debugging code...

Reply:
the "double" code have been put in a single procedure
+ debug-comments is gone
+ checkbox only appears now when there are visited waypoints available

sorry for the comment-overhead, but we are learning from it to adapt to the cgeo-style ;)